spec/money_column_spec.rb

```diff
@@ -36,6 +36,13 @@ class MoneyWithReadOnlyCurrency < ActiveRecord::Base
     expect(record.price).to eq(Money.new(1.23, 'EUR'))
   end
 
+  it 'writes the currency to the db' do
+    expect(Money).to receive(:deprecate).once
+    record.update(price: Money.new(4, 'JPY'))
+    expect(record.price.value).to eq(4)
+    expect(record.price.currency.to_s).to eq('JPY')
```

Comment:
This test doesn't look to confirm the currency is written to the DB, can you load a fresh instance of the object or something after saving?